site/docs/uk/hosting/vps.md

```diff
@@ -52,130 +152,631 @@ systemd --- це потужний менеджер служб, який пере
+# sure your domain's A/AAAA DNS records are properly pointed to
+# this machine's public IP, then replace ":80" below with your
+# domain name.
+
+:80 {
+  # Set this path to your site's directory.
+  root * /usr/share/caddy
+
+  # Enable the static file server.
+  file_server
+
+  # Another common task is to set up a reverse proxy:
+  # reverse_proxy localhost:8080
+
+  # Or serve a PHP site through php-fpm:
+  # php_fastcgi localhost:9000
+}
+
+# Refer to the Caddy docs for more information:
+# https://caddyserver.com/docs/caddyfile
+```
+
+Для роботи бота приведіть конфіг до наступного вигляду:
+
+```text
+<домен> {
+  root * /usr/share/caddy
+  file_server
+  reverse_proxy /<токен-бота> localhost:8000
+}
 ```
 
-### Управління застосунками
+> Замініть `<домен>` на ваш домен, а `<токен-бота>` на токен бота.
 
-#### Старт
+Перезавантажуйте Caddy щоразу, як змінюєте файл конфігураціїї сайту, за допомогою наступної команди:
 
 ```sh
-pm2 start --name <назва-застосунку> <файл-входу>
+systemctl reload caddy
 ```
 
-> `<назва-застосунку>` може бути будь-яким ідентифікатором для вашого застосунку: наприклад, `bot1`.
-> `<файл-входу>` повинен бути шляхом до вашого файлу входу, який запускає вашого бота.
+Тепер всі запити на адресу `https://<домен>/<токен-бота>` будуть переадресовуватися на адресу `http://localhost:8000/<токен-бота>`, на якій працює вебхук бота.
 
-#### Перезапуск
+#### Підключення вебхука до Telegram
 
-Перезапуск застосунку передбачає його зупинку та повторний запуск.
+Залишилося лише сказати Telegram, куди надсилати оновлення.
+Для цього відкрийте браузер й відвідайте сторінку за наступним посиланням:
 
-```sh
-pm2 restart <назва-застосунку>
+```text
+https://api.telegram.org/bot<токен-бота>/setWebhook?url=https://<домен>/<токен-бота>
 ```
 
-#### Перезавантаження
+> Замініть `<токен-бота>` на токен вашого бота, а `<домен>` на ваш домен.
+
+## CI/CD
+
+[CI/CD](https://about.gitlab.com/topics/ci-cd/) --- це важлива складова сучасного процесу розробки програмного забезпечення.
+Цей посібник охоплює майже весь [конвеєр CI/CD](https://about.gitlab.com/topics/ci-cd/cicd-pipeline/).
+
+Ми зосередимося на написанні скриптів для GitHub та GitLab.
+Ви можете легко адаптувати наведені нижче приклади до вашого сервісу CI/CD, наприклад, Jenkins, Buddy тощо, якщо це необхідно.
 
-Перезавантаження застосунку передбачає заміну поточного процесу вашого застосунку на новий, що призводить до відсутності перерви у роботі.
-Це рекомендується для застосунків, які не мають стану.
+:::tip Self-hosted runner
+GitHub та GitLab безкоштовно пропонують певну кількість ресурсів для виконання ваших завдань.
+Однак при налаштуванні конвеєру ви можете швидко використати їх усі, через що вам доведеться платити гроші за додаткові ресурси, інакше завдання не виконуватимуться.
+Щоб такого не сталося, ми рекомендуємо встановити на вашому компʼютері self-hosted runner, щоб завдання виконувалися на вашому компʼютері.
```

Comment:
```suggestion
Щоб такого не сталося, ми рекомендуємо встановити на сервері self-hosted runner, щоб завдання виконувалися на вашому компʼютері.
```

Reply:
This change must be applied first to the original. Although, it is better to leave it as it is and simply indicate that for the test we recommend deploying the runner locally, but for production it is worth considering the option of placing the runner on the server, especially if the development team is running.

Reply:
Overall a bad idea to keep a runner on your computer.